packages/eslint-plugin/src/rules/no-var-requires.ts

```diff
@@ -14,6 +14,7 @@ type MessageIds = 'noVarReqs';
 export default createRule<Options, MessageIds>({
   name: 'no-var-requires',
   meta: {
+    deprecated: true,
```

Comment:
[Bug] Deprecating `no-var-requires` should include removing it from `recommended`. We don't want to be in the business of recommending deprecated rules!

This PR should also add `no-require-imports` to `recommended`.